packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1326,15 +1336,16 @@ export default function buildKernel(
       deliverP = processUpgradeVat(message);
     } else if (message.type === 'changeVatOptions') {
       deliverP = processChangeVatOptions(message);
-    } else if (message.type === 'bringOutYourDead') {
-      deliverP = processBringOutYourDead(message);
     } else if (message.type === 'negated-gc-action') {
       // processGCActionSet pruned some negated actions, but had no GC
       // action to perform. Record the DB changes in their own crank.
     } else if (message.type === 'cleanup-terminated-vat') {
       deliverP = processCleanupTerminatedVat(message);
-    } else if (gcMessages.includes(message.type)) {
-      deliverP = processGCMessage(message);
+    } else if (isGCType(message.type)) {
+      deliverP =
+        message.type === 'bringOutYourDead'
+          ? processBringOutYourDead(message)
```

Comment:
If we left the explicit `else if` case for `bringOutYOuDead` above, typescript isn't smart enough to know that the remaining message type?